CHANGES.md

```diff
@@ -1,6 +1,462 @@
+* fixed support for building our test virtualenv within an existing virtualenv
+* removed our dependency on `subprocess32`
+
+Developer-oriented changes: Module changes
+------------------------------------------
+* removed all instance of `domain._value` from the `LinearAlgebra` module
+
+Developer-oriented changes: Makefile improvements
+-------------------------------------------------
+* improved GASNet build speeds
+* `make` no longer runs GMP and RE2 builds at the same time
+* improved the build configuration of the bundled LLVM sources
+* the bundled LLVM build now sets `GCC_INSTALL_PREFIX` appropriately
+* the bundled LLVM build now builds shared objects by default
+* the bundled LLVM build includes `dump` methods when `CHPL_DEVELOPER` is set
+* enabled asserts in bundled LLVM build when `CHPL_DEVELOPER` or `ASSERTS` set
+* source directories can now contain builds for several architectures
+
+Developer-oriented changes: Compiler Flags
+------------------------------------------
+* added `--library-fortran[-name]` flags in support of Fortran interoperability
+
+Developer-oriented changes: Compiler improvements/changes
+---------------------------------------------------------
+* eliminated all remaining uses of `chpl__tryToken`
+* dramatically improved the representation of forall intents within the IR
+* modernized the representation of reduce expressions
+* improved the creation of promotion iterators and made them more uniform
+* improved the handling of outer variables in for-/forall-expressions
+* fixed the scope resolution of LoopExprs
+* fixed issues with reduce task intents
+* improved the IR representation for default initialization
+* made some internal module params and type aliases available earlier
+* refactored the parser to simplify the declaration of variables
+* made deserialization errors use cnames when compiling in --devel mode
+* variables declared in scopeless blocks now destroyed at end of parent block
+* in module-scope code, call temps are now module-scope variables
+* removed use of `_defaultOf` for numeric types or classes
+* removed `dtSymbol`
+* enabled prototype integration with the Region Vectorizer for LLVM
+* fixed problems with `--llvm-wide-opt` with LLVM 8
+* `--llvm --specialize` compilation generates target metadata for functions
+* migrated `chpl_getPrivatizedCopy()` to module code to improve LLVM performance
+* removed special handling for `cabs()` with `--llvm`
+* `--llvm --gen-ids` compilation now adds LLVM metadata storing Chapel AST ids
+* added experimental support for emitting `llvm.invariant.start`
+* source files supporting `--llvm` are now in `compiler/llvm`
+* `inferConstRefs` now runs on both single-locale and multi-locale compilation
+* removed `reprivatizeIterators`
+
+Developer-oriented changes: Documentation improvements
+------------------------------------------------------
+* converted `CompilerDebugging.txt` to `CompilerDebugging.rst` format
```

Comment:
With the recent change, I'd suggest making this more broad:

* converted several `developer/bestPractices` files to ReStructured Text

Reply:
I just realized that this actually gets pruned from the release.  I'm going to pull it for now but leave the contributor credit in as we discussed, thinking that these should/will be rendered online at some point.